sway/input/cursor.c

```diff
@@ -445,8 +446,13 @@ static void handle_cursor_motion(struct wl_listener *listener, void *data) {
 	}
 
 	wlr_cursor_move(cursor->cursor, event->device, dx, dy);
+
+	// Recalculate pointer location after layout checks
+	node = node_at_coords(cursor->seat,
+		cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
+
 	cursor_send_pointer_motion(cursor, event->time_msec, node, surface,
-		sx + dx, sy + dy);
+		sx, sy);
```

Comment:
This stops confining `s{x,y}`.

Reply:
Could you give an example of how this breaks confinement? My knowledge of the sway code is almost non-existent, so I might be missing something obvious.

Reply:
`wlr_region_confine` will move `sx.sy` into the constrained region. We need to use `sx_confined,sy_confined` somehow. We also need to add `dx,dy` to the cursor position, but adding them to `sx,sy` in the `cursor_send_pointer_motion` seems wrong anyway.

Reply:
At lines 444-445 `dx` and `dy` are updated with `sx_confined,sy_confined`. These are then passed to `wlr_cursor_move` and we pick those new coordinates up in the arguments of `node_at_coords`.

Reply:
Oh, you're right. Forgot `wlr_cursor_move` was doing all the magic.

We should probably add `dx,dy` when calling `node_at_coords` right?

Reply:
From what I can tell the purpose of that code is to get the coordinates of the pointer in the coordinates of the surface to which the constraint is associated. It does that by assuming that the previous position of the cursor is still over the constraining surface, and if not, it bails out. I'm not a fan of that really, I'd rather have it obtain the surface local coordinates in way that doesn't make these assumptions, but couldn't really find one since the constraint doesn't know where the surface is.

Reply:
Okay, that's probably good enough for now.